### PR TITLE
Move modules to make room for "random" AD scheme

### PR DESCRIPTION
**Pull request by rvl**
Relates to #553 

# Overview

- Created modules `Cardano.Wallet.Primitive.Address{Derivation,Discovery}.{Random,Sequential}`
- Left the common code in `Cardano.Wallet.Primitive.Address{Derivation,Discovery}`
- Renamed test modules along the same lines.
- Adjust the rest of the application to import AD.Sequential where it was previously importing the AD module.

# Comments

- `keyToAddress` will have slightly different parameters between Sequential and Random schemes. I haven't done anything to address that, and just left the parameters as they are for Sequential.

### CODE REVIEW

File: lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs [hunk 1]
```diff
@@ -20,15 +20,14 @@
 --
 -- Primitives for performing address derivation for some given schemes. This is
 -- where most of the crypto happens in the wallet and, it is quite important to
--- ensure that the following implementation matches with other wallet softwares
--- (like Yoroi/Icarus or the cardano-cli)
+-- ensure that the implementations match with other Cardano wallets
+-- (like cardano-sl, Yoroi/Icarus, or cardano-cli)
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/AddressDerivation.hs:24]: Updating the module level comments ✨

File: lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs [hunk 2]
```diff
@@ -1,23 +1,18 @@
...
 --
--- This module contains primitives necessary to perform address discovery. So
--- far, we're only interested in address following a sequential derivation
--- scheme as specified in BIP-0044. Later, we may introduce backward
--- compatibility with random address scheme from the legacy Cardano wallets.
+-- This module contains types for address discovery. The two address discovery
+-- schemes implemented are:
+--
+--  * 'Cardano.Wallet.Primitive.AddressDiscovery.Sequential'
+--  * 'Cardano.Wallet.Primitive.AddressDiscovery.Random'
```
Comment (Anviking) [inline lib/core/src/Cardano/Wallet/Primitive/AddressDiscovery.hs:15]: 👍

**Review by Anviking — APPROVED** [review]
Looks good to me! I like the simple interface of `Cardano.Wallet.Primitive.AddressDiscovery` we get now.

This might be a good opportunity to even more critically examine the documentation / structure of code sections etc. I might have a look.

**Comment by rvl** [comment]
Thanks

**Pull request merged**